ZEP000-build-state-on-followers.md

```diff
@@ -0,0 +1,508 @@
+---
+title: Build state on followers
+authors:
+- Christopher Zell (@zelldon)
+  reviewers: []
+  approvers: []
+  editor: TBD
+  creation-date: 2020-04-28
+  last-updated: 2021-06-14
+  status: provisional
+  see-also: []
+  replaces: []
+  superseded-by: []
+---
+
+# Summary
+[summary]: #summary
+
+We should build state on followers to be more reliable, future proven and enable fast fail-over.
+
+# Motivation
+[motivation]: #motivation
+
+Generally we want to reduce the failover time. With that we mean: the time between no leader is available, processing has stopped, and a new leader has been chosen and processing has started again. Especially we want to minimize the impact on process execution latency during failover on a partition. With this ZEP we want to decrease the time from node a becomes Leader until it starts to process new commands. This can be achieved with building state on followers.
```

Comment:
🙃 
```suggestion
Generally we want to reduce the failover time. With that we mean: the time between no leader is available, processing has stopped, and a new leader has been chosen and processing has started again. Especially we want to minimize the impact on process execution latency during failover on a partition. With this ZEP we want to decrease the time from when a node becomes Leader until it starts to process new commands. This can be achieved with building state on followers.
```